ml-conversational-analytic-tool/tests/test_githubDataExtraction.py

```diff
@@ -0,0 +1,483 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import os
+import unittest
+import unittest.mock
+
+import pandas as pd
+import requests
+
+import githubDataExtraction
+
+
+class TestGitHubDataExtraction(unittest.TestCase):
+
+    def setUp(self):
+        self.test_access_token = "test-token"
+        self.test_organization = "test-org"
+        self.test_repo = "test-repo"
+        self.extractor = githubDataExtraction.GithubDataExtractor(self.test_access_token, self.test_organization)
+        self.directory_path = os.path.dirname(os.path.realpath(__file__))
+
+    def test_get_all_repos_returns_list_of_repos(self):
+        repo_one = {"name": "repo_one"}
+        repo_two = {"name": "repo_two"}
+        repo_list = [repo_one, repo_two]
+
+        self.extractor.run_queries = unittest.mock.MagicMock(return_value=repo_list)
+
+        expected = [repo_one.get("name"), repo_two.get("name")]
+        actual = self.extractor.get_all_repos()
+        self.assertEqual(expected, actual)
+
+    def test_get_all_repos_calls_run_queries_with_correct_values(self):
+        repo_one = {"name": "repo_one"}
+        repo_list = [repo_one]
+
+        test_query = """ { test query } """
+        test_queries_and_variables = {
+            "query": test_query,
+            "variables": {"owner": self.test_organization}
+        }
+
+        self.extractor.query_repos = test_query
+        self.extractor.run_queries = unittest.mock.MagicMock(return_value=repo_list)
+
+        self.extractor.get_all_repos()
+
+        self.assertEqual(1, self.extractor.run_queries.call_count)
+        self.extractor.run_queries.assert_called_with(query=test_queries_and_variables)
+
+    def test_get_all_repos_returns_empty_list_when_repos_does_not_exist(self):
+        repo_list = []
+
+        self.extractor.run_queries = unittest.mock.MagicMock(return_value=repo_list)
+
+        actual = self.extractor.get_all_repos()
+        self.assertEqual([], actual)
+
+    def test_get_all_pull_requests_returns_dataframe(self):
+        raw_data_file = open("{}/resources/pr_raw_data_example_one.json".format(self.directory_path))
```

Comment:
The tests should be using "with" statements otherwise the file remains unclosed. Same for all the tests below.

Reply:
Thanks for catching this! all fixed PTAL